packages/mobile/src/geth/saga.ts

```diff
@@ -98,6 +101,76 @@ export function* waitForGethConnectivity() {
   }
 }
 
+function* pollGethStatusUntilSynced() {
+  let head: BlockHeader = yield select(chainHeadSelector)
+
+  // Head may be null on startup
+  if (!head) {
+    head = (yield take(Actions.SET_CHAIN_HEAD)).head
+  }
+
+  while (true) {
+    if (blockIsFresh(head)) {
+      return true
+    }
+
+    head = (yield take(Actions.SET_CHAIN_HEAD)).head
```

Comment:
IIRC the chain head is set to null if Forno is toggled on. We should account for what might happen here if we are in this loop.

Reply:
Yup, I have a check for forno mode upstream in `waitForGethSync` that kicks you out of the flow. I could see it going either place. Do you think it would make more sense in the helper function?